### PR TITLE
Add catalog-related unit tests to demonstrate proposed design

### PR DESCRIPTION
**Pull request by scsewall**

### CODE REVIEW

File: core/backend/src/IModelTransformer.ts [hunk 1]
```diff
@@ -939,11 +944,20 @@ export class TemplateModelCloner extends IModelTransformer {
...
       if (Id64.invalid === this.context.findTargetElementId(predecessorId)) {
-        throw new IModelError(IModelStatus.BadRequest, "Required dependency not found in target iModel", Logger.logError, BackendLoggerCategory.IModelTransformer);
+        if (this.context.isBetweenIModels) {
+          throw new IModelError(IModelStatus.BadRequest, "Required dependency not found in target iModel", Logger.logError, BackendLoggerCategory.IModelTransformer);
+        } else {
+          const definitionElement = this.sourceDb.elements.tryGetElement<DefinitionElement>(predecessorId, DefinitionElement);
+          if (definitionElement && !(definitionElement instanceof RecipeDefinitionElement)) {
+            this.context.remapElement(predecessorId, predecessorId); // when in the same iModel, can use existing DefinitionElements without remapping
+          } else {
+            throw new IModelError(IModelStatus.BadRequest, "Required dependency not found", Logger.logError, BackendLoggerCategory.IModelTransformer);
```
Comment (pmconne) [inline core/backend/src/IModelTransformer.ts:954]: Include `predecessorId` in exception message?

File: core/backend/src/test/standalone/Catalog.test.ts [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+  RecipeDefinitionElement, RepositoryLink, SnapshotDb, SpatialCategory, TemplateModelCloner, TemplateRecipe2d, TemplateRecipe3d,
+  TypeDefinitionElement,
+} from "../../imodeljs-backend";
+import { IModelTestUtils } from "../IModelTestUtils";
+import { IModelTransformerUtils } from "../IModelTransformerUtils";
+import { KnownTestLocations } from "../KnownTestLocations";
+
+const createClassViews = false; // can set to true to make it easier to debug catalog structure
+
+/** Structure of a Catalog
```
Comment (pmconne) [inline core/backend/src/test/standalone/Catalog.test.ts:27]: Does this documentation comment unattached to any type surface somewhere in the published docs?

**Review by pmconne — COMMENTED** [review]
.

File: core/backend/src/IModelTransformer.ts (hunk 1 above)
Comment (scsewall) [reply core/backend/src/IModelTransformer.ts]: Done.

**Review by pmconne — APPROVED** [review]
Looks good to me.

**Review by MichaelBelousov — APPROVED** [review]

File: core/backend/src/test/standalone/Catalog.test.ts [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+  insertEquipmentType(db, containerIdD, "D-101", seriesD1, manufacturerName, productLineNameD);
+  insertEquipmentType(db, containerIdD, "D-102", seriesD1, manufacturerName, productLineNameD);
+  insertEquipmentType(db, containerIdD, "D-201", seriesD2, manufacturerName, productLineNameD);
+  insertEquipmentType(db, containerIdD, "D-202", seriesD2, manufacturerName, productLineNameD);
+
+  db.saveChanges();
+  db.close();
+}
+
+/** This function mocks a Catalog Connector that reads Best Equipment product data and outputs a catalog of components. */
```
[the remark this answers is not in the document]
Comment (scsewall) [reply core/backend/src/test/standalone/Catalog.test.ts]: Yes, fixed.

**Pull request merged**